lib/index.ts

```diff
@@ -67,14 +68,22 @@ class PWMetrics {
 
     for (let runIndex of runs) {
       try {
-        metricsResults[runIndex] = await this.run();
+        const currentMetricResult: MetricsResults = await this.run();
+        currentMetricResult.error = this.resultHasExpectationErrors(currentMetricResult);
```

Comment:
Why not simply declare a boolean outside of the loop? This remove the need to filter the result array in line 81. You could go a step further and check if the boolean has already been set by resultHasExpectationErrors then don't call it again and skip processing.